source/common/stats/symbol_table_impl.cc

```diff
@@ -75,11 +174,87 @@ Symbol SymbolTableImpl::toSymbol(absl::string_view sv) {
   return result;
 }
 
-absl::string_view SymbolTableImpl::fromSymbol(const Symbol symbol) const {
+absl::string_view SymbolTable::fromSymbol(const Symbol symbol) const
+    EXCLUSIVE_LOCKS_REQUIRED(lock_) {
   auto search = decode_map_.find(symbol);
   ASSERT(search != decode_map_.end());
   return search->second;
 }
 
+void SymbolTable::newSymbol() EXCLUSIVE_LOCKS_REQUIRED(lock_) {
+  if (pool_.empty()) {
+    next_symbol_ = ++monotonic_counter_;
+  } else {
+    next_symbol_ = pool_.top();
+    pool_.pop();
+  }
+  // This should catch integer overflow for the new symbol.
```

Comment:
What happens if we do overflow? Can this happen with a very long running server? Sorry I didn't originally review this code so am unfamiliar with the details.

Reply:
nothing good will happen :). But we'd probably also run out of memory before we allocate 4B symbols (say averaging 32 bytes including std::string overhead).

Note that symbols are reference-counted and recycled, so there would have to be 4B *referenced* symbols.